pkg/bmh/list.go

```diff
@@ -19,17 +19,31 @@ const (
 )
 
 // List returns bareMetalHosts inventory in the given namespace.
-func List(apiClient *clients.Settings, nsname string) ([]*BmhBuilder, error) {
-	glog.V(100).Infof("Listing bareMetalHosts in the nsname %s", nsname)
-
+func List(apiClient *clients.Settings, nsname string, options ...goclient.ListOptions) ([]*BmhBuilder, error) {
 	if nsname == "" {
 		glog.V(100).Infof("bareMetalHost 'nsname' parameter can not be empty")
 
 		return nil, fmt.Errorf("failed to list bareMetalHosts, 'nsname' parameter is empty")
 	}
 
+	logMessage := fmt.Sprintf("Listing bareMetalHosts in the namespace %s", nsname)
+	passedOptions := goclient.ListOptions{}
+
+	if len(options) == 1 {
```

Comment:
Hey @ElaiShalevRH , can you rewrite the if blocks as follows:
```
logMessage := fmt.Sprintf("Listing bareMetalHosts in the namespace %s", nsname)
passedOptions := goclient.ListOptions{}

if len(options) > 1 {
    glog.V(100).Infof("'options' parameter must be empty or single-valued")

    return nil, fmt.Errorf("error: more than one ListOptions was passed")
}

if len(options) == 1 {
    passedOptions = options[0]
    logMessage += fmt.Sprintf(" with the options %v", passedOptions)
}
```
I think it is a little bit easier to read